components/switch/src/Switch.js

```diff
@@ -62,14 +48,143 @@ const asArray = val => (Array.isArray(val) ? val : [val]);
+export const Switch = ({
+	name,
+	label,
+	checked: isChecked,
+	onChange,
+	size,
+	block,
+	flipped,
+	toggleText,
+	disabled,
+	srOnlyText,
+}) => {
+	const { COLORS, [pkg.name]: localBrandTokens } = useBrand();
+	const mq = useMediaQuery();
+	const [checked, setChecked] = useState(isChecked);
+	const [switchId] = useState(`switch-${shortid.generate()}`);
+
+	const sizeArr = responsiveMap(asArray(size));
+
+	const localTokens = {
+		labelCss: {},
+		textCss: {},
+		toggleCss: {},
+		toggleTextCss: {},
+	};
+
+	merge(localTokens, localBrandTokens);
 
-export const Switch = ({ size, ...props }) => {
-	const flexiSize = responsiveMap(asArray(size));
+	useEffect(() => {
+		setChecked(isChecked);
+	}, [isChecked]);
+
+	const handleChange = name => () => {
+		if (onChange) {
+			onChange(name);
+		} else {
+			setChecked(!checked);
+		}
+	};
 
 	return (
-		<SwitchContext.Provider value={{ flexiSize }}>
-			<SwitchLabel {...props} />
-		</SwitchContext.Provider>
+		<label
+			htmlFor={switchId}
+			css={mq({
+				display: block ? 'flex' : 'inline-flex',
+				opacity: disabled && 0.5,
+				width: block && '100%',
+				flexWrap: 'wrap',
+				alignItems: 'center',
+				position: 'relative',
+				marginRight: !block && '1.125rem',
+				height: !block && sizeArr.height,
+				marginBottom: '0.375rem',
+				flexDirection: flipped && 'row-reverse',
+				cursor: disabled ? 'not-allowed' : 'pointer',
+				...localTokens.labelCss,
+			})}
+		>
+			<input
+				id={switchId}
+				type="checkbox"
+				name={name}
+				checked={checked}
+				onChange={handleChange(name)}
+				disabled={disabled}
+				css={{
+					position: 'absolute',
+					zIndex: '-1',
+					opacity: 0,
+				}}
+			/>
+			<span
+				css={{
+					flex: block && 1,
+					display: 'flex',
+					alignItems: 'center',
+					whiteSpace: 'normal',
+					position: 'relative',
+					[flipped ? 'paddingLeft' : 'paddingRight']: '0.375rem',
+					...localTokens.textCss,
+				}}
+			>
+				{srOnlyText ? <VisuallyHidden>{label}</VisuallyHidden> : label}
+			</span>
+			<span
+				css={mq({
+					display: 'block',
+					position: 'relative',
+					border: `2px solid ${checked ? COLORS.hero : COLORS.border}`,
+					borderRadius: sizeArr.borderRadius,
+					backgroundColor: checked ? COLORS.hero : '#fff',
+					height: sizeArr.height,
+					width: sizeArr.width,
+					overflow: 'hidden',
+					lineHeight: 1.5,
+					transition: 'border .3s ease,background .3s ease',
+
+					// the thumb/dot
+					'::after': {
+						content: '""',
+						height: sizeArr.height,
+						width: sizeArr.height,
+						display: 'block',
+						position: 'absolute',
+						left: checked ? '100%' : 0,
+						transform: checked ? 'translateX(-100%)' : null,
+						top: 0,
+						borderRadius: '50%',
+						backgroundColor: '#fff',
+						boxShadow: '3px 0 6px 0 rgba(0,0,0,0.3)',
+						transition: 'all .3s ease',
+					},
+					...localTokens.toggleCss,
+				})}
+			>
+				{!!toggleText.length && (
+					<>
```

Comment:
We should use `<Fragment>` to make it explicit and avoid issues with syntax highlighter